shared/src/commonMain/kotlin/org/hyperskill/app/freemium/domain/interactor/FreemiumInteractor.kt

```diff
@@ -22,6 +22,8 @@ class FreemiumInteractor(
 
     suspend fun isProblemsLimitReached(): Result<Boolean> =
```

Comment:
I think that we can use `SubscriptionType.areProblemLimitsEnabled` here instead of check freemium subscription to hide details about what subscriptions have limits in `SubscriptionType`